backend/tests/test_advanced_features.py

```
@@ -0,0 +1,490 @@
+
+This module tests advanced broadcasting features including:
+- Message priority handling
+- Message persistence for offline clients
+- Rate limiting to prevent client overload
+- Message compression for large payloads
+
+Following Canon TDD methodology - implement one test at a time.
+"""
+
+import pytest
+import asyncio
```

Ruff (0.12.2)

`asyncio` imported but unused

Remove unused import: `asyncio`

(F401)